org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CallHierarchyHandler.java

```diff
@@ -192,9 +205,25 @@ private IMember getCallHierarchyElement(String uri, int line, int character, IPr
 			if (member != null) {
 				incomingMethodWrapperCache.put(member, call);
 			}
-			CallHierarchyItem symbol = toCallHierarchyItem(call.getMember());
-			List<Range> ranges = toCallRanges(call.getMethodCall().getCallLocations());
-			result.add(new CallHierarchyIncomingCall(symbol, ranges));
+			Collection<CallLocation> callLocations = call.getMethodCall().getCallLocations();
+			List<Range> ranges = toCallRanges(callLocations);
```

Comment:
move down the if block

Reply:
not fixed. ranges should only be computed if callLocations != null

Reply:
Fixed.

Reply:
No. still not fixed. `List<Range> ranges = toCallRanges(callLocations);` needs to go inside the if block.
```java
Collection<CallLocation> callLocations = call.getMethodCall().getCallLocations();
if (callLocations != null) {
	List<Range> ranges = toCallRanges(callLocations);
	for (CallLocation location : callLocations) {
		IOpenable openable = getOpenable(location);
		Range callRange = getRange(openable, location);
		CallHierarchyItem symbol = toCallHierarchyItem(call.getMember());
		symbol.setSelectionRange(callRange);
		result.add(new CallHierarchyIncomingCall(symbol, ranges));
	}
}
```